dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -464,13 +464,36 @@
 		<IlcCompileDependsOn>Compile;_ComputeLinkerArguments;_ComputeManagedAssemblyToLink;SetupOSSpecificProps;PrepareForILLink;_XamarinComputeIlcCompileInputs</IlcCompileDependsOn>
 	</PropertyGroup>
 
+	<!--
+		Workaround for SDK issue where setting PublishAot=true during build cannot resolve correct
+		runtime pack and IL compiler pack.
+
+		Setting PublishAotUsingRuntimePack=true unconditionally results in the runtime pack label to
+		be updated to "NativeAOT" instead of "Mono" and trying to build against NativeAOT runtime
+		pack instead of the Mono one.
+
+		Setting PublishAotUsingRuntimePack=false causes the NuGet resolution to check for known
+		ILCompiler packs and these don't exist for iOS-like platforms.
+
+		The workaround ensure that PublishAotUsingRuntimePack is false when SDK updates the runtime
+		pack labels, and thus the "Mono" label is used. We then set it to true right before
+		ProcessFrameworkReferences target to skip downloading the target ILCompiler pack.
+	-->
+	<Target Name="_WorkaroundAotRuntimePackResolution" BeforeTargets="ProcessFrameworkReferences" Condition="'$(PublishAot)' == 'true' And '$(_UseNativeAot)' != 'true'">
+		<PropertyGroup>
+			<PublishAotUsingRuntimePack>true</PublishAotUsingRuntimePack>
+		</PropertyGroup>
+	</Target>
+
```

Comment:
Correct fix here: https://github.com/dotnet/sdk/pull/34640

If it's accepted and ingested then we should remove this workaround and unconditionally set `<PublishAotUsingRuntimePack>true</PublishAotUsingRuntimePack>` when `PublishAot=true` in dotnet/targets/Xamarin.Shared.Sdk.props.

Reply:
Filed to https://github.com/xamarin/xamarin-macios/issues/18741 to not forget to remove the workaround.